### PR TITLE
Handle failure on add filter modal

### PR DESCRIPTION
**Pull request by Swiip**
Fix https://github.com/ProtonMail/proton-mail-settings/issues/197

Handle request failure in filter creation. Close the modal and update the list.

### CODE REVIEW

File: containers/filters/AddFilterModal.js [hunk 1]
```diff
@@ -38,12 +38,15 @@ function AddFilterModal({ filter, type, mode = 'create', onEdit = noop, ...props
...
-            text: c('Notification').t`${Filter.Name} created`
-        });
-        props.onClose();
+        try {
+            const { Filter } = await reqCreate.request(filter);
+            createNotification({
+                text: c('Notification').t`${Filter.Name} created`
+            });
+        } finally {
+            call();
```
Comment (EpokK) [inline containers/filters/AddFilterModal.js]: If the request fail we should not call the event manager since there is no filter model to synchronize.

Comment (Swiip) [reply containers/filters/AddFilterModal.js]: It's what I thought at the beginning but in fact yes.

In fact, the filter is added but not activated and you're not allow to activate it. But it's added so without that, it's not shown.

Comment (EpokK) [reply containers/filters/AddFilterModal.js]: It sounds like a bug, you should check the issue with @bartbutler.

Comment (Swiip) [reply containers/filters/AddFilterModal.js]: I just checked with @bartbutler.

It works as expected. The limit is on the number of active filters not "existing" filters. This way, someone spending time to configure a new filter doesn't loose his work.

Comment (EpokK) [reply containers/filters/AddFilterModal.js]: @Swiip could you add a comment in the code explaining why we are calling `call` from the event manager even if the request fail please?

**Pull request merged**